pkgs/tools/X11/xprompt/default.nix

```diff
@@ -0,0 +1,56 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, writeText
+, fontconfig
+, libX11
+, libXft
+, libXinerama
+, conf ? null
+, nix-update-script
+}:
+
+stdenv.mkDerivation rec {
+  pname = "xprompt";
+  version = "2.5.0";
+
+  src = fetchFromGitHub {
+    owner = "phillbush";
+    repo = "xprompt";
+    rev = "v${version}";
+    sha256 = "sha256-pOayKngUlrMY3bFsP4Fi+VsOLKCUQU3tdkZ+0OY1SCo=";
+  };
+
+  buildInputs = [
+    fontconfig
+    libX11
+    libXft
+    libXinerama
+  ];
+
+  postPatch = with lib;
+    let
+      configFile =
+        if isDerivation conf || builtins.isPath conf
```

Comment:
I suggest to use `lib.isDerivation` directly here.

Reply:
Why? I don't see any issues with scope shadowing here.

Reply:
Just a suggestion, nothing binding.

I prefer to be explicit because `with` expressions can complicate a static code analysis.

Reply:
I see, I'll keep that in mind.